flake.nix

```diff
@@ -24,372 +23,81 @@
-                      --set NIX_YOSYS_PLUGIN_DIRS $out/share/yosys/plugins \
-                      ${module_flags}
-                  '';
-                });
+            doCheck = true; # FIXME(ac): can we turn these back on?
 
+            passthru = {
+              inherit (prev) withPlugins;
               allPlugins = { ghdl = yosys-ghdl; };
-            in { inherit withPlugins allPlugins; };
-
-            meta = with lib; {
-              description = "Open RTL synthesis framework and tools";
-              homepage = "https://yosyshq.net/yosys/";
-              license = licenses.isc;
-              platforms = platforms.all;
-              maintainers = with maintainers; [ thoughtpolice ];
             };
-          };
-
-        nextpnr-xilinx = with final;
-          stdenv.mkDerivation rec {
-            pname = "nextpnr-xilinx";
-            version = "0.5.0";
-
-            srcs = [
-              (fetchgit {
-                url = "https://github.com/openXC7/nextpnr-xilinx";
-                rev = version;
-                fetchSubmodules = true;
-                deepClone = false;
-                hash = "sha256-mDYEmq3MW1kK9HeR4PyGmKQnAzpvlOf+H66o7QTFx3k=";
-                leaveDotGit = false;
-              })
-            ];
-
-            sourceRoot = "nextpnr-xilinx";
+          })).override { inherit abc-verifier; };
 
-            nativeBuildInputs = [ cmake git ];
-            buildInputs = [ python310Packages.boost python310 eigen ]
-              ++ (lib.optional stdenv.cc.isClang llvmPackages.openmp);
+          nextpnr-xilinx = callPackage ./nix/nextpnr-xilinx.nix { };
 
-            setupHook = ./nextpnr-setup-hook.sh;
+          prjxray = callPackage ./nix/prjxray.nix { };
 
-            cmakeFlags = [
-              "-DCURRENT_GIT_VERSION=${
-                lib.substring 0 7 (lib.elemAt srcs 0).rev
-              }"
-              "-DARCH=xilinx"
-              "-DBUILD_GUI=OFF"
-              "-DBUILD_TESTS=OFF"
-              "-DUSE_OPENMP=ON"
-            ];
-
-            installPhase = ''
-              mkdir -p $out/bin
-              cp nextpnr-xilinx bba/bbasm $out/bin/
-              mkdir -p $out/usr/share/nextpnr/external
-              cp -rv ../xilinx/external/prjxray-db $out/usr/share/nextpnr/external/
-              cp -rv ../xilinx/external/nextpnr-xilinx-meta $out/usr/share/nextpnr/external/
-              cp -rv ../xilinx/python/ $out/usr/share/nextpnr/python/
-              cp ../xilinx/constids.inc $out/usr/share/nextpnr
-            '';
-
-            doCheck = false;
-
-            meta = with lib; {
-              description = "Place and route tool for FPGAs";
-              homepage = "https://github.com/openXC7/nextpnr-xilinx";
-              license = licenses.isc;
-              platforms = platforms.all;
-              maintainers = with maintainers; [ thoughtpolice ];
+          nextpnr-xilinx-chipdb = {
+            artiz7 = callPackage ./nix/nextpnr-xilinx-chipdb.nix {
```

Comment:
It is artix7, not artiz7

Reply:
Sorry we noticed you fixed this in your repo but haven't pulled those changes in yet, we will rebase as soon as a decision is made on https://github.com/openXC7/toolchain-nix/pull/5.